### PR TITLE
Scheduler course batch

### PR DESCRIPTION
**Pull request by sudhirgiri2911**
PR for scheduling course batch count

### CODE REVIEW

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 1]
```diff
@@ -25,6 +27,11 @@
  */
 public class ManageCourseBatchCount implements Job {
 
+  private Map<String, Map<String, Object>> contentMap = new HashMap<>();
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: do not use states in job -> ideally job should be stateless.

Use variable passing - if you require - but initalize this at start of execute and as soon as execute method is over - this should get garbage collected

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 2]
```diff
@@ -25,6 +27,11 @@
...
+  private Map<String, List<Map<String, Object>>> openBatchMap = new HashMap<>();
+  private Map<String, List<Map<String, Object>>> privateBatchMap = new HashMap<>();
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: true for all class members - avoid declaring class members

File: actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/BulkUploadManagementActor.java [hunk 3]
```diff
@@ -140,7 +140,7 @@ private void getBulkUploadDownloadStatusLink(Request actorMessage) {
                 cloudStorageData.getFileName());
         Response response = new Response();
         response.setResponseCode(ResponseCode.OK);
-        Map<String,Object> resultMap  = response.getResult();
+        Map<String, Object> resultMap = response.getResult();
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/BulkUploadManagementActor.java]: irrelevant change  - please revert

**Comment by amolnin** [comment]
Merege the code properly - few irrelevant files are showing up.
Clean-up and push - i will review - ping once done

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 4]
```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
 
     return info;
   }
+
+  private void updateCourseBatchMap(
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: updateCourseBatchStatus

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 5]
```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+  }
+
+  private void getContentAndStore(Map<String, List<Map<String, Object>>> batchMap) {
+    Map<String, String> ekstepHeader = CourseBatchSchedulerUtil.headerMap;
+    for (Map.Entry<String, List<Map<String, Object>>> batchList : batchMap.entrySet()) {
+      String courseId = batchList.getKey();
+      if (!contentMap.containsKey(courseId)) {
+        Map<String, Object> ekStepContent =
+            CourseEnrollmentActor.getCourseObjectFromEkStep(courseId, ekstepHeader);
+        if (null != ekStepContent && ekStepContent.size() > 0) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: MapUtils.isNotEmpty

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 6]
```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+    contentMap.get(courseId).put(contentName, val);
+  }
+
+  private void getContentAndStore(Map<String, List<Map<String, Object>>> batchMap) {
+    Map<String, String> ekstepHeader = CourseBatchSchedulerUtil.headerMap;
+    for (Map.Entry<String, List<Map<String, Object>>> batchList : batchMap.entrySet()) {
+      String courseId = batchList.getKey();
+      if (!contentMap.containsKey(courseId)) {
+        Map<String, Object> ekStepContent =
+            CourseEnrollmentActor.getCourseObjectFromEkStep(courseId, ekstepHeader);
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: Can be more efficient if - multiple courseIds can be fetched together...
can you confirm if that is possible.

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 7]
```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+    }
+    if (isCountDecrementStatus) {
+      courseBatchMap.put(JsonKey.COUNTER_DECREMENT_STATUS, true);
+      courseBatchMap.put(JsonKey.COUNT_DECREMENT_DATE, todayDate);
+      courseBatchMap.put(JsonKey.STATUS, ProjectUtil.ProgressStatus.COMPLETED.getValue());
+    }
+    courseBatchMap.put(JsonKey.UPDATED_DATE, ProjectUtil.getFormattedDate());
+  }
+
+  private void updateBatchMap(Map<String, Object> map) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: updateCourseIdToBatchListMap

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 8]
```diff
@@ -65,31 +72,33 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
-          weakMap.put(JsonKey.COURSE_ID, (String) map.get(JsonKey.COURSE_ID));
-          weakMap.put(JsonKey.COUNTER_DECREMENT_STATUS, true);
-          weakMap.put(JsonKey.COUNT_DECREMENT_DATE, ProjectUtil.getFormattedDate());
-          weakMap.put(JsonKey.STATUS, ProjectUtil.ProgressStatus.COMPLETED.getValue());
-          weakMap.put(JsonKey.UPDATED_DATE, ProjectUtil.getFormattedDate());
-          weakMap.put(JsonKey.HASHTAGID, (String) map.get(JsonKey.HASHTAGID));
-          CourseBatchSchedulerUtil.updateCourseBatchDbStatus(weakMap, false);
+          updateBatchMap(map);
+        }
+      }
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: above 10 lines are repeated twice - once for start-date & once for end-date
Cn be made common - and processed 1-by-1 invoking the function call -> with passing arguments for variability

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 9]
```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+    } else {
+      val -= size;
+      if (val < 0) {
+        val = 0;
+      }
+    }
+    contentMap.get(courseId).put(contentName, val);
+  }
+
+  private void getContentAndStore(Map<String, List<Map<String, Object>>> batchMap) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: updateCourseDetailsMap

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 10]
```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
 
     return info;
   }
+
+  private void updateCourseBatchMap(
+      boolean isCountIncrementStatus,
+      boolean isCountDecrementStatus,
+      Map<String, Object> courseBatchMap) {
+    String todayDate = new SimpleDateFormat("yyyy-MM-dd").format(new Date());
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: LocalDateTime.now()
       .format(DateTimeFormatter.ofPattern(""yyyy-MM-dd")

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java (hunk 4 above)
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: private function -> only invocation passes both values false?
when are we expecting call to function with true?

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 11]
```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+        privateBatchMap.get(courseId).add(map);
+      } else {
+        List<Map<String, Object>> batchList = new ArrayList<>();
+        batchList.add(map);
+        privateBatchMap.put(courseId, batchList);
+      }
+    }
+  }
+
+  private void contentMapUpdate(boolean increment) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: updateAllCourseBatchCount

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 12]
```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+      if (response) {
+        openBatchMap
+            .get(courseId)
+            .forEach(
+                map -> {
+                  if (CourseBatchSchedulerUtil.updateDataIntoES(map)) {
+                    CourseBatchSchedulerUtil.updateDataIntoCassandra(map);
+                  }
+                });
+        contentName = CourseBatchSchedulerUtil.getContentName(JsonKey.INVITE_ONLY);
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: again similar repetion - veriability is invite_only vs open -rest of the code is pretty much the same.

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 13]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
-    }
-    Map<String, String> ekstepHeader = getBasicHeader();
-    // addHeaderProps(ekstepHeader, JsonKey.CHANNEL_ID , hashTagId);
-    // collect data from EKStep.
+    String contentName = getContentName(enrollmentType);
+    boolean response = false;
     Map<String, Object> ekStepContent =
-        CourseEnrollmentActor.getCourseObjectFromEkStep(courseId, ekstepHeader);
+        CourseEnrollmentActor.getCourseObjectFromEkStep(courseId, getBasicHeader());
     if (ekStepContent != null && ekStepContent.size() > 0) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: MapUtils.isNotEmpty

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 14]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
-      if (increment) {
-        val = val + 1;
-      } else {
-        if (val != 0) val = val - 1;
-      }
+      int val = getUpdatedBatchCount(ekStepContent, contentName, increment);
       if (ekStepContent.get(JsonKey.CHANNEL) != null) {
         ProjectLogger.log(
-            "Channel value is coming from Contnet "
+            "Channel value is coming from Content "
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: Channel value coming from content is "

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 15]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
                 + (String) ekStepContent.get(JsonKey.CHANNEL)
                 + " Id "
                 + courseId,
             LoggerEnum.INFO.name());
         addHeaderProps(
-            ekstepHeader, JsonKey.CHANNEL_ID, (String) ekStepContent.get(JsonKey.CHANNEL));
+            getBasicHeader(), JsonKey.CHANNEL_ID, (String) ekStepContent.get(JsonKey.CHANNEL));
       } else {
         ProjectLogger.log(
             "Channel value is  not coming from Contnet Id " + courseId, LoggerEnum.INFO.name());
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: "No channel value available in content with Id "

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 16]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
+    String response = "";
+    try {
+      ProjectLogger.log("updating content details to Ekstep start", LoggerEnum.INFO.name());
+      String contentUpdateBaseUrl = ProjectUtil.getConfigValue(JsonKey.EKSTEP_BASE_URL);
+      response =
+          HttpUtil.sendPatchRequest(
+              contentUpdateBaseUrl
+                  + PropertiesCache.getInstance().getProperty(JsonKey.EKSTEP_CONTENT_UPDATE_URL)
+                  + courseId,
+              "{\"request\": {\"content\": {\"" + contentName + "\": " + val + "}}}",
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: create map - use ObjectMapper to convert to json.

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 17]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
+              getBasicHeader());
+      ProjectLogger.log(
+          "batch count update response==" + response + " " + courseId, LoggerEnum.INFO.name());
+    } catch (IOException e) {
+      ProjectLogger.log("Error while updating content value " + e.getMessage(), e);
+    }
+    if (response.equalsIgnoreCase(JsonKey.SUCCESS)) return true;
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: return JsonKey.SUCCESS.equalsIgnoreCase(response)

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 18]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
+    } else {
+      contentName = "c_" + name + "_private_batch_count";
+    }
+    return contentName;
+  }
+
+  public static int getUpdatedBatchCount(
+      Map<String, Object> ekStepContent, String contentName, boolean increment) {
+    int val = (int) ekStepContent.getOrDefault(contentName, 0);
+    if (increment) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: val = increment?val+1:(val>0)?val-1:0

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 19]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
+              + dto.getAdditionalProperties().toString(),
+          LoggerEnum.INFO.name());
+    }
+    return listOfMap;
+  }
+
+  public static String getContentName(String enrollmentType) {
+    String name = ProjectUtil.getConfigValue(JsonKey.SUNBIRD_INSTALLATION);
+    String contentName = "";
+    if (enrollmentType.equals(ProjectUtil.EnrolmentType.open.getVal())) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: reverse the condition - constant on left side - value on right side

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 20]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
     }
-    return JsonKey.SUCCESS.equalsIgnoreCase(response);
+    return response;
   }
 
   private static Map<String, String> getBasicHeader() {
     return headerMap;
   }
+
+  private static List<Map<String, Object>> courseBatchStartAndCompleteData(
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: getToBeUpdatedCourseList

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 21]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
+    Map<String, Object> map = new HashMap<>();
+    Map<String, String> dateRangeFilter = new HashMap<>();
+    dateRangeFilter.put("<=", date);
+    map.put(dateAttribute, dateRangeFilter);
+    map.put(counterAttribute, false);
+    dto.addAdditionalProperty(JsonKey.FILTERS, map);
+    return contentReadResponse(dto);
+  }
+
+  private static List<Map<String, Object>> courseBatchStartStatusData(String startDate) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: getToBeUpdatedCourseListByDateAndStatus

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 22]
```diff
@@ -42,63 +42,25 @@ private CourseBatchSchedulerUtil() {}
...
-        listOfMap = (List<Map<String, Object>>) val;
-      } else {
-        ProjectLogger.log(
-            "No data found for start date course batch===" + startDate, LoggerEnum.INFO.name());
-      }
-    } else {
-      ProjectLogger.log(
-          "No data found for start date course batch===" + startDate, LoggerEnum.INFO.name());
+    List<Map<String, Object>> courseBatchStartedList =
+        courseBatchStartAndCompleteData(startDate, true);
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: getToBeUpdatedCoursesByDate

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java (hunk 20 above)
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: getToBeUpdatedCoursesByDate

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java (hunk 21 above)
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: getToBeUpdatedCoursesByStatus

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 23]
```diff
@@ -110,4 +115,136 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+    if (batchMap.containsKey(courseId)) {
+      batchMap.get(courseId).add(map);
+    } else {
+      List<Map<String, Object>> batchList = new ArrayList<>();
+      batchList.add(map);
+      batchMap.put(courseId, batchList);
+    }
+  }
+
+  private void courseDetailsMapUpdate(boolean increment) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: updateAllCourseBatchCount

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 24]
```diff
@@ -110,4 +115,136 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
 
     return info;
   }
+
+  private void performAction(boolean increment) {
+    updateCourseDetailsMap(openBatchMap);
+    updateCourseDetailsMap(privateBatchMap);
+    courseDetailsMapUpdate(increment);
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: handleUpdateBatchCount

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 25]
```diff
@@ -110,4 +115,136 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
 
     return info;
   }
+
+  private void performAction(boolean increment) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: handleUpdateBatchCount

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 26]
```diff
@@ -110,4 +115,136 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+              "ManagerCourseBatchCount: updateDetailsMap: No content Found for courseId "
+                  + courseId,
+              LoggerEnum.INFO);
+        }
+      }
+    }
+  }
+
+  private void updateEkstepAndDb() {
+    for (Map.Entry<String, Map<String, Object>> ekStepUpdateMap : courseDetailsMap.entrySet()) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: ekStepUpdateMap -> courseDetails

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 27]
```diff
@@ -110,4 +115,136 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+          ProjectLogger.log(
+              "ManagerCourseBatchCount: updateDetailsMap: No content Found for courseId "
+                  + courseId,
+              LoggerEnum.INFO);
+        }
+      }
+    }
+  }
+
+  private void updateEkstepAndDb() {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: doUpdateCourseBatchCount

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java [hunk 28]
```diff
@@ -110,4 +115,136 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
...
+    for (Map.Entry<String, Map<String, Object>> ekStepUpdateMap : courseDetailsMap.entrySet()) {
+      String courseId = ekStepUpdateMap.getKey();
+      updateESAndCassandra(
+          openBatchMap.get(courseId), ekStepUpdateMap.getValue(), courseId, JsonKey.OPEN);
+      updateESAndCassandra(
+          privateBatchMap.get(courseId), ekStepUpdateMap.getValue(), courseId, JsonKey.INVITE_ONLY);
+    }
+  }
+
+  private void updateESAndCassandra(
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java]: doUpdateCourseBatchCount

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 29]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
+    dateRangeFilter.put("<=", startDate);
+    map.put(JsonKey.START_DATE, dateRangeFilter);
+    map.put(JsonKey.COUNTER_INCREMENT_STATUS, true);
+    map.put(JsonKey.STATUS, 0);
+    dto.addAdditionalProperty(JsonKey.FILTERS, map);
+    return contentReadResponse(dto);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static List<Map<String, Object>> contentReadResponse(SearchDTO dto) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: searchContent

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 30]
```diff
@@ -161,68 +128,135 @@ private static void addHeaderProps(Map<String, String> header, String key, Strin
...
+    } else {
+      ProjectLogger.log(
+          "CourseBatchSchedulerUtil: contentReadResponse: No response from ElasticSearch for "
+              + dto.getAdditionalProperties().toString(),
+          LoggerEnum.INFO.name());
+    }
+    return listOfMap;
+  }
+
+  public static String getContentName(String enrollmentType) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: getCountName

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java (hunk 29 above)
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: searchContent

**Pull request merged**